app/lib/email.server.ts

```diff
@@ -89,6 +91,40 @@ function maybeThrowSES(e: any, warning: string) {
   ])
 }
 
+/**
+ * Amazon SES has a byte limit of 262144 for TemplateData arguments.
+ * If the circular body is over this size limit, the message will be truncated.
+ * If the body is truncated, the circular website link will be removed.
+ * A warning that the email has been truncated is added and the circular website link is added back in.
+ * An additional buffer is allowed for for variation in url length and truncation warning.
+ */
+export function truncateEmailBodyIfNecessary({
+  body,
+  circularId,
+  origin,
+}: {
+  body: string
+  circularId?: number
+  origin?: string
+}) {
+  const textBody = getBody(body)
+  const inputBytes = Buffer.byteLength(textBody, 'utf-8')
+
+  if (inputBytes <= 262144) return textBody
+
+  const buffer = Buffer.from(textBody, 'utf-8')
+  const truncatedBuffer = buffer.subarray(0, 261900)
```

Comment:
I truncated it generously here to allow a fair bit of margin. We can adjust this. I thought it wouldn't matter if it was truncated to be exactly 262144 bytes at the end. If we do want it to be at the exact maximum, I would get the byte size of the truncation warning message, then I would subtract that from 262144 and make the subarray that size, but I figured that just giving it a wide margin and not bothering with that complexity was likely fine.